### PR TITLE
Update dune dependency and dune-file

### PR DESCRIPTION
**Pull request by lessp**

### CODE REVIEW

**Comment by lessp** [comment]
> Why ? Adding name. Why removing format ? (It is by default)

The new `dune-project` is generated by dune, it defaults to those values.

File: package.json [hunk 1]
```diff
@@ -14,17 +14,15 @@
...
-      "esy-installer harfbuzz.install"
     ]
   },
   "dependencies": {
-    "@opam/dune": ">=1.6.0",
     "@esy-ocaml/reason": "^3.3.7",
-    "ocaml": ">=4.4.0",
-    "esy-harfbuzz": "^1.9.1005"
+    "@opam/dune": ">=2.0.1",
+    "@opam/dune-configurator": ">=2.0.1",
```
Comment (Et7f3) [inline package.json:22]: Theese constraints allow dune 2.1 and 2.2 to be installed and thoses version don't read dune lang 2.3 so dune-project file should be modified.

File: package.json [hunk 2]
```diff
@@ -14,17 +14,15 @@
...
+    "esy-harfbuzz": "^1.9.1005",
+    "ocaml": ">=4.4.0"
   },
-  "devDependencies": {
-  }
+  "devDependencies": {}
```
Comment (Et7f3) [inline package.json]: Can be removed... because empty.

Comment (lessp) [reply package.json]: Not related to this PR, but I'll remove it!

File: package.json (hunk 1 above)
Comment (lessp) [reply package.json:22]: Resolved in: 6faf802

File: package.json (hunk 2 above)
Comment (lessp) [reply package.json]: Resolved in: 6faf802

**Review by Et7f3 — APPROVED** [review]

**Comment by bryphe** [comment]
This PR isn't usable from Revery or Onivim 2:
```
info install 0.6.2 (using package.json)
info checking https://github.com/ocaml/opam-repository for updates...
info checking https://github.com/esy-ocaml/esy-opam-override for updates...
info resolving esy packages: done                         
info solving esy constraints: done         
error: No solution found:

Conflicting constraints:
  revery -> flex -> @opam/dune -> ocaml@>=4.2.3000 && ocaml@<4.8.0
  revery -> ocaml@=4.8.1000

Conflicting constraints:
  revery -> flex -> @opam/dune@>=opam:1.0.0 <opam:2.0.0
  revery -> flex -> @opam/dune@>=opam:1.0.0 <opam:2.0.0

Conflicting constraints:
  revery -> flex -> @opam/dune@>=opam:1.0.0 <opam:2.0.0
  revery -> reason-harfbuzz -> @opam/dune@>=opam:2.0.1
 ```

**Comment by bryphe** [comment]
I can't bring it in as a resolution. So unfortunately I have to revert.

**Comment by lessp** [comment]
> I can't bring it in as a resolution. So unfortunately I have to revert.

Ah, sorry about that! I think flex is actually updated but needs an npm-release!

**Pull request merged**